### ISSUE TITLE
False positive in case of custom __setattr__ that changes the type of the attribute

### ISSUE THREAD

**Issue by sjdv1982** [python/mypy#13071]
**Bug Report**

I have a class with a custom `__setattr__(self,  attr, value:TYPE)`. This method converts `value` to another type.
I am getting false positives, since mypy assumes that for each particular `TYPE`, `type(getattr(obj, attr)) == TYPE`. 

**To Reproduce**
Minimal code example:
```python
from typing import *

class Context:
    def __setattr__(self, attr:str, value:Any):
        super().__setattr__(attr, str(value))

ctx = Context()
ctx.a = 10
print(len(ctx.a))
```
**Actual Behavior**

mypy output
```
....py:9: error: "Context" has no attribute "a"
....py:9: error: Argument 1 to "len" has incompatible type "int"; expected "Sized"
```
I would like to tell mypy that for this class, it mustn't assume that type(getattr(ctx, "a")) == type(10) . 

**Your Environment**

- mypy 0.961 (compiled: yes) on Ubuntu Linux  (Jupyter Docker image)
- Python 3.8.10 (Anaconda)
